addons/info/src/index.js

```diff
@@ -83,12 +67,29 @@ export function addInfo(storyFn, context, info, _options) {
   );
 }
 
-export const withInfo = (info, _options) =>
-  addonCompose((storyFn, context) => addInfo(storyFn, context, info, _options));
+export const withInfo = textOrOptions => {
+  if (typeof textOrOptions === 'string') {
+    const text = textOrOptions;
+    return withInfo({ text });
+  }
+  const options = textOrOptions;
+  return storyFn => context => addInfo(storyFn, context, options);
+};
+
+export { Story };
 
 export default {
-  addWithInfo: deprecate(function addWithInfo(storyName, info, storyFn, _options) {
-    return this.add(storyName, withInfo(info, _options)(storyFn));
+  addWithInfo: deprecate(function addWithInfo(storyName, text, storyFn, options) {
+    if (typeof storyFn !== 'function') {
```

Comment:
Could use a bit of refactoring,
in some cases the const `text` is actually the `storyFn`. I find this confusing.

I'd prefer using abstract names or simply an argument array to do detection, and only assign to a const when we are sure what it is.